Code/Framework/AzCore/AzCore/Serialization/SerializeContext.h

```diff
@@ -2516,7 +2512,7 @@ namespace AZ
         using GenericClassInfoType = typename SerializeGenericTypeInfo<T>::ClassInfoType;
         static_assert(AZStd::is_base_of<AZ::GenericClassInfo, GenericClassInfoType>::value, "GenericClassInfoType must be be derived from AZ::GenericClassInfo");
 
-        const AZ::TypeId& canonicalTypeId = AzTypeInfo<T>::Uuid();
+        const AZ::TypeId& canonicalTypeId = GenericClassInfoType().GetSpecializedTypeId();
```

Comment:
The AssetSerializer<T> template is a special case when it comes to it's SpecializedTypeId.
It doesn't return the actual Id of the instantiated type, but instead the generic [AssetClassId](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Asset/AssetSerializer.h#L121-L124)

Reply:
Should we make note of this somewhere, to clarify what's going on?